### PR TITLE
Node.js CLI Input

### LINKED ISSUE #1306: Node.js CLI Input
## Brief Summary
In this tutorial, we will learn how to accept input from the user in a Node.js CLI application using the `readline` package.

## Key Takeaways

- Set up the `readline` object.
- Learn how to ask a question and use a callback function to process the result from the user.
- Write an `async` function with an `asyncIterator` to simulate halting the flow of the program to get input.

### PR DESCRIPTION
**Pull request by zolomohan**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.

**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process and style-once published, you can then submit any number of articles.)**
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

This fixes #1306

### CODE REVIEW

**Comment by adrianmurage** [comment]
@hectorkambow 
kindly run a plagiarism check

**Comment by hectorkambow** [comment]
[Node.js CLI Input #1342.pdf](https://github.com/section-io/engineering-education/files/5860672/Node.js.CLI.Input.1342.pdf)

Plagiairism check done - @zolomohan  The % is a bit higher than usual - please review and make sure to add any needed sources and or edit 🚀

**Comment by zolomohan** [comment]
@hectorkambow @adrianmurage I edited some lines to avoid plagiarism.

File: articles/nodejs-cli-input/index.md [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+author: mohan-raj
+date: 2021-01-14T00:00:00-18:00
+topics: [Node.js]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/nodejs-cli-input/hero.jpg
+    alt: Node.js CLI Input Hero Image
+---
+In this tutorial, we will learn how to get a user input in a Node.js CLI application. To do this, you'll need to listen to `STDIN` (Standard Input), which Node.js exposes as `process.stdin`, a readable stream.
```
Comment (adrianmurage) [inline articles/nodejs-cli-input/index.md:17]:
```suggestion
In this tutorial, we will learn how to get user input in a Node.js CLI application. To do this, you'll need to listen to `STDIN` (Standard Input), which Node.js exposes as `process.stdin`, a readable stream.
```

File: articles/nodejs-cli-input/index.md [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+<!--more-->
+
+Streams are used to deal with I/O. You can learn more about it in this [documentation](https://nodejs.org/api/stream.html).
+
+### Prerequisites
+
+- You should have a basic understanding of the JavaScript language.
```
Comment (adrianmurage) [inline articles/nodejs-cli-input/index.md:24]:
```suggestion
- You should have an understanding of the JavaScript language.
```
Using the term "basic" might discourage someone who is struggling to understand the concepts you've written about.

**Review by adrianmurage — CHANGES_REQUESTED** [review]
Great article @zolomohan 
I've left you a few changes to make.
Ping me when you're done.

**Comment by zolomohan** [comment]
@adrianmurage I've applied those suggestions.

**Review by adrianmurage — APPROVED** [review]

**Comment by adrianmurage** [comment]
@hectorkambow 
This article is ready for final review

**Pull request merged**